src/python/pants/engine/isolated_process.py

```diff
@@ -16,23 +16,24 @@
 logger = logging.getLogger(__name__)
 
 
-class ExecuteProcessRequest(datatype(['argv', 'env', 'input_files_digest', 'digest_length'])):
+class ExecuteProcessRequest(datatype(['argv', 'env', 'input_files_digest', 'digest_length', 'output_files'])):
```

Comment:
As mentioned on the other PR, I feel about 75% strongly that this should support globs. I could be convinced otherwise, but my understanding of why the "known output files" restriction exists elsewhere is that it's purely to allow for applicative builds. And it would immediately introduce a bunch of otherwise unnecessary logic for codegen.

Reply:
The remote execution API supports capturing individual files, and capturing whole directories, but nothing in between. My plan is to add directory capturing when we have a concrete use case for it.

If we find this isn't flexible enough, we should draft a change to the API, and standardise it with folks, but I'm hoping we can get by with just files and directories.

Reply:
Oof. Yea, standardizing glob matching would be a challenge. Darn.